ui-admin/src/study/surveys/FormOptions.tsx

```diff
@@ -0,0 +1,163 @@
+import InfoPopup from '../../components/forms/InfoPopup'
+
+/** component for selecting versions of a form */
+export default function FormOptions({
+  studyEnvContext, workingForm, updateWorkingForm, onDismiss, isDirty,
+  visibleVersionPreviews, setVisibleVersionPreviews, isConsentForm = false
+}:
+                                          {studyEnvContext: StudyEnvContextT, workingForm: VersionedForm,
+                                            isDirty: boolean,
+                                            updateWorkingForm: (props: SaveableFormProps) => void,
+                                            onDismiss: () => void
+                                            visibleVersionPreviews: VersionedForm[],
+                                            setVisibleVersionPreviews: (versions: VersionedForm[]) => void,
+                                          isConsentForm?: boolean}) {
+  const stableId = workingForm.stableId
+
+  const downloadJSON = () => {
+    const content = workingForm.content
+    // To get this formatted nicely and not as one giant line, need to parse
+    // this as an object and then stringify the result.
+    const blob = new Blob(
+      [JSON.stringify(JSON.parse(content), null, 2)],
+      { type: 'application/json' })
+    const filename = `${stableId}_v${workingForm.version}${isDirty ? '_draft' : ''}.json`
+    saveBlobAsDownload(blob, filename)
+  }
+  const isSurvey = !!(workingForm as Survey).surveyType
+
+  return <Modal show={true} onHide={onDismiss}>
+    <Modal.Header closeButton>
+      <Modal.Title>{workingForm.name} <InfoPopup placement="bottom" content={<div>
+        See the &quot;Options&quot; section in
+        our <ZendeskLink doc={DocsKey.SURVEY_EDIT}>survey editing docs</ZendeskLink> for
+        more detail.  Note that you need to  &quot;Save&quot; the survey
+        for any changes to options on this page to take effect.
+      </div>}/></Modal.Title>
+    </Modal.Header>
+    <Modal.Body>
+      { isSurvey &&
+        <label className="form-label d-block">
+          <input type="checkbox" checked={(workingForm as Survey).required}
+            onChange={e => updateWorkingForm({
+              ...workingForm, required: e.target.checked
+            })}
+          /> Required
+        </label>
+      }
+      <Button variant="primary" onClick={onDismiss}>
+        Ok
+      </Button>
+      <hr/>
+      <Button variant="secondary" title="Download the current contents of the JSON Editor as a file"
+        onClick={downloadJSON}>
+        <FontAwesomeIcon icon={faDownload}/> Download JSON
+      </Button>
+      <hr/>
+      <VersionSelector studyEnvContext={studyEnvContext}
+        workingForm={workingForm}
+        visibleVersionPreviews={visibleVersionPreviews}
+        setVisibleVersionPreviews={setVisibleVersionPreviews}
+        isConsentForm={!isSurvey}
+        onDismiss={onDismiss}/>
+    </Modal.Body>
+  </Modal>
+}
+
+
+/**
+ *
+ */
+export const VersionSelector = ({
```

Comment:
Every time i previewed something on the version selector it disappeared

Reply:
I believe that's intentional (I agree it's confusing).  The idea is that you can't open a preview of something you're already previewing.  In any case, it predates this change.

Reply:
Hmm, yeah I am finding this whole flow (editing, saving, etc.) very confusing but that is totally outside of the scope of this PR.